gui/account/forms.py

```diff
@@ -320,6 +324,20 @@ def clean_bsdusr_home(self):
 
             if home.startswith('/mnt/'):
                 bsdusr_username = self.cleaned_data.get('bsdusr_username', '')
+                volumes = ['/mnt/{}'.format(volume.vol_name) for volume in Volume.objects.all()]
+
+                if self.instance.id:
+                    if home != user_home:
+                        if len(zfs.list_datasets(path=user_home)) > 0 and \
+                                (len(zfs.list_datasets(path=home)) > 0 and home.startswith(user_home + '/')):
+                            raise forms.ValidationError(_("A dataset inside the home dataset "
+                                                          "cannot be used as a home directory."))
+
+                    else:
```

Comment:
I dont think we should do an `else`. It is common to not change user directory, and that is allowed.

`if self.instance.id and if home != user_home:`